src/handlers/int-test-support/helpers/s3Helper.ts

```diff
@@ -93,45 +118,82 @@ const putS3ObjectBasic = async (
 
 const putS3Object = callWithRetryAndTimeout(putS3ObjectBasic);
 
-const deleteS3ObjectBasic = async (
-  object: S3Object
-): Promise<DeleteObjectCommandOutput> => {
+const deleteS3ObjectsBasic = async (
+  params: deleteS3ObjectByPrefixes
+): Promise<DeleteObjectsCommandOutput> => {
   if (runViaLambda())
     return (await sendLambdaCommand(
-      IntTestHelpers.deleteS3Object,
-      object
-    )) as unknown as DeleteObjectCommandOutput;
-
-  const bucketParams = {
-    Bucket: object.bucket,
-    Key: object.key,
+      IntTestHelpers.deleteS3Objects,
+      params
+    )) as unknown as DeleteObjectsCommandOutput;
+  const result: DeleteObjectsCommandOutput = {
+    Deleted: [],
+    Errors: [],
+    $metadata: {},
   };
-  return await s3Client.send(new DeleteObjectCommand(bucketParams));
+  for (const prefixToDelete of params.prefixesToDelete) {
+    const listResult = await listS3Objects({
+      bucketName: params.bucketName,
+      prefix: prefixToDelete,
+    });
+    if (listResult.Contents && listResult.Contents.length > 0) {
+      const keysToDelete = listResult.Contents.map(({ Key }) => ({ Key }));
+      const deleteResult = await deleteS3Object({
+        bucket: params.bucketName,
+        keysToDelete,
+      });
+      if (deleteResult.Deleted) {
+        result.Deleted?.push(...deleteResult.Deleted);
+      }
+      if (deleteResult.Errors) {
+        result.Errors?.push(...deleteResult.Errors);
+      }
+    }
+  }
+  return result;
 };
 
-const deleteS3Object = callWithRetryAndTimeout(deleteS3ObjectBasic);
+const deleteS3Objects = callWithRetryAndTimeout(deleteS3ObjectsBasic);
 
-const deleteS3Objects = async (
-  params: BucketAndPrefix
-): Promise<DeleteObjectCommandOutput[]> => {
+/* Deletes s3 objects by keys in batches */
+const deleteS3ObjectBasic = async (
+  params: deleteS3ObjectByKeys
+): Promise<DeleteObjectsCommandOutput> => {
   if (runViaLambda())
     return (await sendLambdaCommand(
-      IntTestHelpers.deleteS3Objects,
+      IntTestHelpers.deleteS3Object,
       params
-    )) as unknown as DeleteObjectCommandOutput[];
-
-  const result = await listS3Objects(params);
-
-  if (result.Contents === undefined) return [];
-
-  return await Promise.all(
-    result.Contents.map(
-      async (item) =>
-        await deleteS3Object({ bucket: params.bucketName, key: item.Key ?? "" })
-    )
-  );
+    )) as unknown as DeleteObjectsCommandOutput;
+  const result: DeleteObjectsCommandOutput = {
```

Comment:
Same with the AWS type bleeding into our code here.

Reply:
Agreed to do the type changes in separate PR as the changes will be more need to update all the calling functions which uses listS3Objects and DeleteS3Objects